pkg/providers/azure/vm.go

```diff
@@ -51,6 +51,12 @@ func (d *vmProvider) GetResource(ctx context.Context) (*schema.Resources, error)
 				for _, ipConfig := range ipconfigList {
 					privateIP := *ipConfig.PrivateIPAddress
 
+					// The PublicIPAddress field can be nil especially for SQL Server
+					// VMs that are pulled.
+					// In these cases, we just want to return an empty error
+					if ipConfig.PublicIPAddress == nil {
+						return nil, nil
```

Comment:
this should probably `continue`, instead of returning as we do want to process other vms in the loop

Reply:
I had tried with continue, but never received results back. Not sure why that's occurring but when returning nil,nil I receive results back.

Reply:
Resolved. Had to add another check later on for more nil

Reply:
Resolved. Had to add another check for nil later on.